engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/preparation/JobPreparationActor.scala

```diff
@@ -0,0 +1,233 @@
+      else {
+        dockerHashingActor ! data.dockerHashRequest
+        stay() using data.newAttempt(newRequestLostTimer)
+      }
+  }
+  
+  whenUnhandled {
+    case Event(unexpectedMessage, _) =>
+      workflowLogger.error(s"JobPreparation actor received an unexpected message in state $stateName: $unexpectedMessage")
+      stay()
+  }
+  
+  private [preparation] def evaluateInputsAndAttributes = {
+    for {
+      evaluatedInputs <- resolveAndEvaluateInputs(jobKey, workflowDescriptor, expressionLanguageFunctions, executionData.outputStore)
+      runtimeAttributes <- prepareRuntimeAttributes(evaluatedInputs)
+    } yield (evaluatedInputs, runtimeAttributes)
+  }
+  
+  private def startPreparation(inputs: Map[Declaration, WdlValue], attributes: Map[LocallyQualifiedName, WdlValue]) = {
+    def sendDockerRequest(dockerImageId: DockerImageIdentifierWithoutHash) = {
+      val dockerHashRequest = DockerHashRequest(dockerImageId, dockerHashCredentials)
+      val newData = JobPreparationActorData(dockerHashRequest, inputs, attributes, newRequestLostTimer, maxDockerAttempts)
+      dockerHashingActor ! dockerHashRequest
+      goto(WaitingForDockerHash) using Option(newData)
+    }
+    
+    def handleDockerValue(value: String) = DockerImageIdentifier.fromString(value) match {
+      case Success(dockerImageId: DockerImageIdentifierWithoutHash) => sendDockerRequest(dockerImageId)
+      case Success(dockerImageIdWithHash: DockerImageIdentifierWithHash) =>
+        // If the docker value already has a hash - we're ok for call caching
+        val response = prepareBackendDescriptor(inputs, attributes, CallCachingEligible)
+        sendResponseAndStop(response)
+      case Failure(failure) => sendFailureAndStop(failure)
+    }
+
+    attributes.get(RuntimeAttributesKeys.DockerKey) match {
+      case Some(dockerValue) => handleDockerValue(dockerValue.valueString)
+      case None =>
+        // If there is no docker attribute at all - we're ok for call caching
+        val response = prepareBackendDescriptor(inputs, attributes, CallCachingEligible)
+        sendResponseAndStop(response)
+    }
+  }
+  
+  // Schedule a DockerNoResponseTimeout message to be sent to self
+  private final def newRequestLostTimer = context.system.scheduler.scheduleOnce(dockerNoResponseTimeout, self, DockerNoResponseTimeout)(ec, self)
+  // Schedule a DockerHashRequest to be sent to the docker hashing actor
+  private final def newBackPressureTimer(dockerHashRequest: DockerHashRequest) = context.system.scheduler.scheduleOnce(backpressureWaitTime, dockerHashingActor, dockerHashRequest)(ec, self)
+
+  /*
+   * If we get backpressured, recreate both timers
+   */
+  private def handleDockHashBackPressure(request: DockerHashRequest, data: JobPreparationActorData) = {
+    val requestLostTimer = newRequestLostTimer
+    val backpressureTimer = newBackPressureTimer(data.dockerHashRequest)
```

Comment:
The `backpressure` method on the FSM data called right after cancels them. But I think you're right it's actually better to cancel them explicitly before creating the new ones 👍